File: frontend/src/api/discovery.ts
import { apiClient } from './client';

export interface DiscoveryResult {
  user: {
    settings: {
      id: string;
      name: string;
      scope: string;
      source: string;
      path: string;
      exists: boolean;
      editable: boolean;
    };
    skills: ResourceItem[];
  };
  project: {
    settings: {
      id: string;
      name: string;
      scope: string;
      source: string;
      path: string;
      local_path: string;
      exists: boolean;
      local_exists: boolean;
      editable: boolean;
    };
    skills: ResourceItem[];
  };
  managed: {
    id: string;
    name: string;
    scope: string;
    source: string;
    editable: boolean;
    settings: Record<string, unknown>;
    mcp: Record<string, unknown>;
    extra: Record<string, unknown>[];
    warnings: string[];
  };
  plugin_cache: ResourceItem[];
  cli: {
    id: string;
    name: string;
    scope: string;
    source: string;
    installed: boolean;
    version: string | null;
    editable: boolean;
  };
  summary: {
    total_skills: number;
    total_plugin_cache: number;
    editable_count: number;
    readonly_count: number;
  };
}

export interface ResourceItem {
  id: string;
  name: string;
  scope: string;
  source: string;
  path: string;
  editable: boolean;
  version?: string;
}

export const discoveryApi = {
  /** Discover all Claude Code resources */
  getAll: (projectRoot?: string) => {
    const params = projectRoot ? `?project_root=${encodeURIComponent(projectRoot)}` : '';
    return apiClient.get<DiscoveryResult>(`/discovery/${params}`);
  },

  /** Get Claude CLI status */
  getCliStatus: () =>
    apiClient.get<DiscoveryResult['cli']>('/discovery/cli'),

  /** Get managed configuration (read-only) */
  getManaged: () =>
    apiClient.get<DiscoveryResult['managed']>('/discovery/managed'),

  /** List user-level skills */
  getUserSkills: () =>
    apiClient.get<ResourceItem[]>('/discovery/user/skills'),

  /** List project-level skills */
  getProjectSkills: (projectRoot?: string) => {
    const params = projectRoot ? `?project_root=${encodeURIComponent(projectRoot)}` : '';
    return apiClient.get<ResourceItem[]>(`/discovery/project/skills${params}`);
  },

  /** List plugin cache entries (read-only) */
  getPluginCache: () =>
    apiClient.get<ResourceItem[]>('/discovery/plugin-cache'),
};

File: backend/api/discovery.py
"""Discovery API — resource discovery across all scopes."""

from fastapi import APIRouter, Query
from typing import Optional

from services import discovery_service

router = APIRouter()


@router.get("/")
async def discover_resources(project_root: Optional[str] = Query(None)):
    """Discover all Claude Code resources.

    Returns a unified view of:
      - user: settings, skills
      - project: settings, local settings, skills
      - managed: managed settings/MCP (read-only)
      - plugin_cache: cached plugins (read-only)
      - cli: Claude CLI status
      - summary: counts
    """
    return discovery_service.discover_all(project_root)


@router.get("/cli")
async def discover_cli_status():
    """Get Claude CLI installation status."""
    return discovery_service.discover_cli_status()


@router.get("/managed")
async def discover_managed():
    """Get managed configuration (read-only)."""
    return discovery_service.discover_managed()


@router.get("/user/skills")
async def discover_user_skills():
    """List user-level skills."""
    return discovery_service.discover_user_skills()


@router.get("/project/skills")
async def discover_project_skills(project_root: Optional[str] = Query(None)):
    """List project-level skills."""
    return discovery_service.discover_project_skills(project_root)


@router.get("/plugin-cache")
async def discover_plugin_cache():
    """List plugin cache entries (read-only)."""
    return discovery_service.discover_plugin_cache()
